cpp/pathwidth/Pathwidth.cpp

```diff
@@ -0,0 +1,63 @@
+#include "pathwidth/Pathwidth.hpp"
+
+namespace Koala {
+    NetworKit::count Pathwidth::SubtreeSize(NetworKit::count n, NetworKit::count v) {
+        if (recognition->cotree->nodes[v].type == 2) {
+            recognition->cotree->nodes[v].size = 1;
+            return 1;
+        } else {
+            NetworKit::count l = 0, r = 0;
+            if (recognition->cotree->nodes[v].left_son != -1) {
+                l = SubtreeSize(n, recognition->cotree->nodes[v].left_son);
+            }
+
+            if (recognition->cotree->nodes[v].right_son != -1) {
+                r = SubtreeSize(n, recognition->cotree->nodes[v].right_son);
+            }
+            recognition->cotree->nodes[v].size = r + l + 1;
+            return r + l + 1;
+        }
+    }
+
+    NetworKit::count Pathwidth::pathwidth(NetworKit::count n, NetworKit::count v) {
+        if (recognition->cotree->nodes[v].type == 2) {
```

Comment:
Again, extracting `recognition->cotree->nodes[v]` to an autoref variable would simplify your code a lot.